pennylane/labs/tests/trotter_error/product_formulas/test_pt_integration.py

```diff
@@ -21,75 +21,44 @@
+core_constant, one_body, two_body = qml.qchem.electron_integrals(mol)()
 
-two_chem = 0.5 * qml.math.swapaxes(two_body, 1, 3)  # V_pqrs
+two_chem = qml.math.swapaxes(two_body, 1, 3)  # V_pqrs
 one_chem = one_body - 0.5 * qml.math.einsum("pqss", two_body)  # T_pq
 
-core_shift, one_shift, two_shift = qml.qchem.symmetry_shift(
-    nuc_core, one_chem, two_chem, n_elec=mol.n_electrons
-)  # symmetry-shifted terms of the Hamiltonian
-
-_, two_body_cores, two_body_leaves = qml.qchem.factorize(
-    two_shift, tol_factor=1e-2, cholesky=True, compressed=True, regularization="L2"
-)  # compressed double-factorized shifted two-body terms with "L2" regularization
-
-two_core_prime = qml.math.eye(mol.n_orbitals) * two_body_cores.sum(axis=-1)[:, None, :]
-one_body_extra = qml.math.einsum(
-    "tpk,tkk,tqk->pq", two_body_leaves, two_core_prime, two_body_leaves
-)  # one-body correction
-
-one_body_eigvals, one_body_eigvecs = qml.math.linalg.eigh(one_shift + one_body_extra)
-one_body_cores = qml.math.expand_dims(qml.math.diag(one_body_eigvals), axis=0)
-one_body_leaves = qml.math.expand_dims(one_body_eigvecs, axis=0)
-cdf_hamiltonian = {
-    "nuc_constant": core_shift[0],
-    "core_tensors": qml.math.concatenate((one_body_cores, two_body_cores), axis=0),
-    "leaf_tensors": qml.math.concatenate((one_body_leaves, two_body_leaves), axis=0),
-}  # CDF Hamiltonian
-
-circ_wires = range(2 * mol.n_orbitals)
-hf_state = qml.qchem.hf_state(electrons=mol.n_electrons, orbitals=len(circ_wires))
-
-
-@qml.qnode(qml.device("default.qubit", wires=circ_wires))
-def create_state():
-    """Create a basis state"""
-    qml.BasisState(hf_state, wires=circ_wires)
-    return qml.state()
-
-
-state = create_state()
-h = 0.1
-
-# Load the CDF integrals
-U = np.array(cdf_hamiltonian["leaf_tensors"][1:])
-Z = np.array(cdf_hamiltonian["core_tensors"][1:])
-U0 = np.array(cdf_hamiltonian["leaf_tensors"][0])
-Z0 = np.array(cdf_hamiltonian["core_tensors"][0])
-
-# Recreate the fragments
-two_body_fragments = np.einsum("tpk,tqk,tkl,trl,tsl->tpqrs", U, U, Z, U, U)
-one_body = U0 @ Z0 @ U0.T
-
-fermionic_fragments = [fermionic_observable(one=one_body, constant=[0.0]).to_mat(format="dense")]
-for i in range(two_body_fragments.shape[0]):
-    obc = qml.math.einsum("pk,kk,qk->pq", U[i], Z[i], U[i])
-    fermionic_fragments.append(
-        fermionic_observable(one=-obc, two=two_body_fragments[i], constant=[0.0]).to_mat(
-            format="dense"
-        )
-    )
+_, two_body_cores, two_body_leaves = qml.qchem.factorize(two_chem, tol_factor=1e-2)
+
+one_body_eigvals, one_body_eigvecs = qml.math.linalg.eigh(one_chem)
+one_body_cores = qml.math.diag(one_body_eigvals)
+one_body_leaves = one_body_eigvecs
+
+# CDF Hamiltonian
+U0 = one_body_leaves
+Z0 = one_body_cores
+U = two_body_leaves
+Z = two_body_cores
+
+eri = np.einsum("tpk,tqk,tkl,trl,tsl->tpqrs", U, U, Z, U, U)  # regenerate V_pqrs
+h1e = U0 @ Z0 @ U0.T
+h1e = h1e + 0.5 * qml.math.einsum("pqss", two_body)  # regenerate h_pq
```

Comment:
Why subtract and add `qml.math.einsum("pqss", two_body)` when we really don't use the intermediate value?